lib/ansible/plugins/strategy/__init__.py

```diff
@@ -507,8 +507,13 @@ def parent_handler_match(target_handler, handler_name):
                         self._tqm._stats.increment('changed', original_host.name)
                 self._tqm.send_callback('v2_runner_on_failed', task_result, ignore_errors=ignore_errors)
             elif task_result.is_unreachable():
-                self._tqm._unreachable_hosts[original_host.name] = True
-                iterator._play._removed_hosts.append(original_host.name)
+                ignore_unreachable = original_task.ignore_unreachable
+                if not ignore_unreachable:
+                    self._tqm._unreachable_hosts[original_host.name] = True
+                    iterator._play._removed_hosts.append(original_host.name)
```

Comment:
does a task result ever get registered for when not ignoring unreachable?

Reply:
It would be treated the same way as it was, so I think the answer is no.

Reply:
tempted to 'fix' that in the future, but not something this PR should handle

Reply:
I'm not sure how that would be used, since unreachable host errors out without `ignore_unreachable`, the playbook wouldn't continue to use he registered result. Or do you mean a different "register" than I'm thinking?

Reply:
I meant to change it to allow it to register, though it would only be visible via `hostvars[<unreachablehost>][<regvar>]`